examples/shiftbot/src/map.ts

```diff
@@ -0,0 +1,227 @@
+            map_without_doors.foreground[x][map.height - 3] = new Block('gravity_up')
+            map_without_doors.foreground[x][map.height - 2] = new Block('gravity_up')
+            map_without_doors.foreground[x + 1][map.height - 2] = new Block('hazard_stripes')
+        }
+    }
+
+    // Right door
+    for (let y = map.height - 1; y >= 1; y--) {
+        if (map_without_doors.foreground[map.width - 1][y].name == 'key_door_green') {
+            map_without_doors.foreground[map.width - 1][y] = new Block('gravity_left')
+            map_without_doors.foreground[map.width - 2][y] = new Block('gravity_left')
+            map_without_doors.foreground[map.width - 3][y] = new Block('gravity_left')
+            map_without_doors.foreground[map.width - 4][y] = new Block('gravity_left')
+            map_without_doors.foreground[0][y + 1] = new Block('hazard_stripes')
+        }
+    }
+
+    // Top Door
+    for (let x = map.width - 1; x >= 0; x--) {
+        if (map_without_doors.foreground[x][1].name == 'key_door_green') {
+            map_without_doors.foreground[x][0] = new Block('gravity_down')
+            map_without_doors.foreground[x][1] = new Block('gravity_down')
+            map_without_doors.foreground[x][2] = new Block('gravity_down')
+            map_without_doors.foreground[x][3] = new Block('gravity_down')
+            map_without_doors.foreground[x - 1][0] = new Block('hazard_stripes')
+        }
+    }
+
+    // Left Door
+    for (let y = 1; y < map.height - 1; y++) {
+        if (map_without_doors.foreground[1][y].name == 'key_door_green') {
+            map_without_doors.foreground[0][y] = new Block('gravity_right')
+            map_without_doors.foreground[1][y] = new Block('gravity_right')
+            map_without_doors.foreground[2][y] = new Block('gravity_right')
+            map_without_doors.foreground[3][y] = new Block('gravity_right')
+            map_without_doors.foreground[0][y + 1] = new Block('hazard_stripes')
+        }
+    }
+}
+
+function construct_map() {
+    const map_no_border = get_map()
+    console.log(map_no_border.meta) // TODO keep till bug gets fixed
+    map.paste(1, 1, map_no_border)
+    map_without_doors.paste(1, 1, map_no_border)
+    encode_doors()
+    return map_no_border.meta
+}
+
+function find_map(search_string: string): [string, string] | null {
+    let maps: [string, { name: string }][] = fs.readdirSync(MAPS_PATH)
+        .filter(p => p.startsWith('map'))
+        .map(name => [name, Structure.metaFromString(
+            fs.readFileSync(path.join(MAPS_PATH, name))
+                .toString('ascii'))])
+    const map = maps.find(p => p[1].name == search_string)
+    if (map) return [map[0], map[1].name]
+    maps = maps.filter(p => p[1].name.includes(search_string))
+    if (maps.length == 0) return null
+    return [maps[0][0], maps[0][1].name]
+}
+
+export async function open_door() {
+    await client.block(TOP_LEFT.x + 22, TOP_LEFT.y + map.height - 2, 1, 'gravity_right')
```

Comment:
```
const dx = [22, 26, 28, 29]
// for loop...
```

Reply:
Done in https://github.com/Anatoly03/pixelwalker.js/pull/18/commits/e345ea8f8c8bfec90694dbf1e844b301da36cf30